libpod/image.go

```diff
@@ -0,0 +1,229 @@
+	// being written.  Except in specific circumstances, no value should be
+	// specified, indicating that the shared, system-wide default policy
+	// should be used.
+	SignaturePolicyPath string
+	// ReportWriter is an io.Writer which will be used to log the writing
+	// of the new image.
+	ReportWriter io.Writer
+	// Store is the local storage store which holds the source image.
+	Store storage.Store
+	// DockerRegistryOptions encapsulates settings that affect how we
+	// connect or authenticate to a remote registry to which we want to
+	// push the image.
+	common.DockerRegistryOptions
+	// SigningOptions encapsulates settings that control whether or not we
+	// strip or add signatures to the image when pushing (uploading) the
+	// image to a registry.
+	common.SigningOptions
+}
+
+// Image API
+
+// ImageFilter is a function to determine whether an image is included in
+// command output. Images to be outputted are tested using the function. A true
+// return will include the image, a false return will exclude it.
+type ImageFilter func(*storage.Image) bool
+
+// PullImage pulls an image from configured registries
+// By default, only the latest tag (or a specific tag if requested) will be
+// pulled. If allTags is true, all tags for the requested image will be pulled.
+// Signature validation will be performed if the Runtime has been appropriately
+// configured
+func (r *Runtime) PullImage(imgName string, allTags bool) error {
+	// PullImage copies the image from the source to the destination
+	var (
+		images []string
+	)
+
+	srcRef, err := alltransports.ParseImageName(imgName)
+	if err != nil {
+		defaultName := DefaultRegistry + imgName
+		srcRef2, err2 := alltransports.ParseImageName(defaultName)
+		if err2 != nil {
+			return errors.Errorf("error parsing image name %q: %v", defaultName, err2)
+		}
+		srcRef = srcRef2
+	}
+
+	splitArr := strings.Split(imgName, ":")
+
+	// supports pulling from docker-archive, oci, and registries
+	if splitArr[0] == "docker-archive" {
+		tarSource := tarfile.NewSource(splitArr[len(splitArr)-1])
+		manifest, err := tarSource.LoadTarManifest()
+		if err != nil {
+			return errors.Errorf("error retrieving manifest.json: %v", err)
+		}
+		// to pull all the images stored in one tar file
+		for i := range manifest {
+			if manifest[i].RepoTags != nil {
+				images = append(images, manifest[i].RepoTags[0])
+			} else {
+				// create an image object and use the hex value of the digest as the image ID
+				// for parsing the store reference
+				newImg, err := srcRef.NewImage(r.imageContext)
+				if err != nil {
+					return err
+				}
+				defer newImg.Close()
+				digest := newImg.ConfigInfo().Digest
+				if err := digest.Validate(); err == nil {
+					images = append(images, "@"+digest.Hex())
+				} else {
+					return errors.Wrapf(err, "error getting config info")
+				}
+			}
+		}
+	} else if splitArr[0] == "oci" {
+		// needs to be implemented in future
+		return errors.Errorf("oci not supported")
```

Comment:
Didn't the pull request to add this get merged?

Reply:
Correction, that's still waiting to be merged - #830 

This should go in after that and pick up the changes from it